messaging/src/main/java/org/axonframework/eventhandling/gateway/DefaultEventGateway.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2010-2019. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.gateway;
+
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.MessageDispatchInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+/**
+ * Default implementation of the EventGateway interface. It allow configuration of {@link MessageDispatchInterceptor EventDispatchInterceptors}.
+ * The Event Dispatch Interceptors can intercept and alter events dispatched on this specific gateway. Typically,
+ * this would be used to add gateway specific meta data to the Event.
+ *
+ * @author Bert laverman
+ * @since 4.0.4
+ */
+public class DefaultEventGateway extends AbstractEventGateway implements EventGateway {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultEventGateway.class);
+
+    private DefaultEventGateway(Builder builder) {
+        super(builder);
+    }
+
+    /**
+     * Return a builder.
```

Comment:
I'd suggest to flesh out this bit of javadoc a bit, something like:
```
/**
 * Instantiate a Builder to be able to create a {@link DefaultEventGateway}.
 * <p>
 * The {@code dispatchInterceptors} are defaulted to an empty list.
 * The {@link EventBus} is a <b>hard requirement</b> and as such should be provided.
 *
 * @return a Builder to be able to create a {@link DefaultEventGateway}
 */
```

Reply:
Added everything _but_ the info on defaults and requirements, as that has little to do with `builder()`. Put those lines instead in the class javadocs as noted below.